docs/products/postgresql/concepts/pg-connection-pooling.rst

```diff
@@ -0,0 +1,92 @@
+
+Why connection pooling?
+------------------------
+
+A high number of backend connections can become a problem with PostgreSQL, as the resource cost per connection is quite high due to how PostgreSQL manages client connections. PostgreSQL creates a separate backend process for each connection, and the unnecessary memory usage caused by the processes will start affecting the total throughput of the system at some point. Moreover, if each connection is very active, the performance can be affected by the high number of parallel executing tasks.
+
+It makes sense to have enough connections so that each CPU core on the server has something to do (each connection can only utilise a single CPU core), but a hundred connections per CPU core may be too much. All this is workload-specific, but often a good number of connections to have is roughly 3-5 times the CPU core count.
+
+
+
+.. Note::
+    PostgreSQL 9.6 introduced limited parallelization support for `running queries in parallel <https://www.postgresql.org/docs/9.6/static/parallel-query.html>`_ on multiple CPU cores.
+
+
+Without a connection pooler, the database connections are handled directly by PostgreSQL backend processes, with one process per connection:
+
+.. mermaid::
+
+    graph LR
+        classDef style_main_node fill:orange,stroke:#333,stroke-width:0px;spacing:0px;
+        classDef style_pg_services fill:#90ee90,stroke:#333,stroke-width:0px;
+        classDef style_client fill:lightgray,stroke:#333,stroke-width:0px;
+        pg_client_1(PG client) <-.->|Client establishing a new connection| postmaster
+        pg_client_2(PG client) ---> pg_backend_1(PG Backend 1)
+        pg_client_3(PG client) ---> pg_backend_2(PG Backend 2)
+        pg_client_4(PG client) ---> pg_backend_3(PG Backend 3)
+        pg_client_5(PG client) --->|Existing client connections| pg_backend_4(PG Backend 4)
+        pg_client_6(PG client) ---> pg_backend_5(PG Backend 5)
+        pg_client_7(PG client) ---> pg_backend_6(PG Backend 6)
+
+    subgraph PostgreSQL server
+        postmaster & pg_backend_1 & pg_backend_2 & pg_backend_3 & pg_backend_4 & pg_backend_5 & pg_backend_6
+    end
+
+    subgraph Clients
+        pg_client_1 & pg_client_2 & pg_client_3 & pg_client_4 & pg_client_5 & pg_client_6 & pg_client_7
+    end
+    class pg_client_1 style_main_node;
+    class pg_backend_1,pg_backend_2,pg_backend_3,pg_backend_4,pg_backend_5,pg_backend_6,postmaster style_pg_services;
+    class Clients style_client;
+
+Adding a PGBouncer pooler that utilizes fewer backend connections frees up server resources for more important uses, such as disk caching:
```

Comment:
Could we get a few more words how this diagram illustrates your point, I'm not quite sure what I'm looking at.

Reply:
Added a note